core/data/src/test/java/com/google/samples/apps/nowinandroid/core/data/testdoubles/TestNewsResourceDao.kt

```diff
@@ -67,6 +67,22 @@ class TestNewsResourceDao : NewsResourceDao {
                 result
             }
 
+    override fun getNewsResourceIdsByFilter(
+        useFilterTopicIds: Boolean,
+        filterTopicIds: Set<String>,
+        useFilterNewsIds: Boolean,
+        filterNewsIds: Set<String>,
+    ): Flow<List<String>> {
+        val filteredNewsResourceIds = entitiesStateFlow.map { newsResourceEntities ->
+            newsResourceEntities.filter { entity ->
+                !useFilterTopicIds || topicCrossReferences.any {
+                    it.newsResourceId == entity.id && it.topicId in filterTopicIds
+                } && (!useFilterNewsIds || entity.id in filterNewsIds)
```

Comment:
Removing the parentheses changes the meaning here; I believe you need to retain them for the intended behavior.

Reply:
You're right, I'm reverting it back.

Reply:
@